src/main/java/org/opensearch/security/privileges/PrivilegesEvaluationContext.java

```diff
@@ -172,4 +180,41 @@ public String toString() {
             + mappedRoles
             + '}';
     }
+
+    public Permissions getPermissionsForApiToken() {
+        return permissionsForApiToken;
+    }
+
+    @VisibleForTesting
+    PrivilegesEvaluationContext(
+        User user,
+        ImmutableSet<String> mappedRoles,
+        String action,
+        ActionRequest request,
+        Task task,
+        IndexResolverReplacer indexResolverReplacer,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Supplier<ClusterState> clusterStateSupplier,
+        Permissions permissions
+    ) {
+        this.user = user;
+        this.mappedRoles = mappedRoles;
+        this.action = action;
+        this.request = request;
+        this.clusterStateSupplier = clusterStateSupplier;
+        this.indexResolverReplacer = indexResolverReplacer;
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+        this.task = task;
+        this.permissionsForApiToken = permissions;
+    }
+
+    private Permissions extractApiTokenPermissionsForUser(ApiTokenRepository apiTokenRepository) {
```

Comment:
Can this call be moved to within the PrivilegesEvaluator and then pass the resulting permissions into this context?

Since there is overlap here between roles/token permissions I'm wondering if this can be structured in a way to make it obvious that privileges are being evaluated on a user that is mapped to roles or on a token that has a set of permissions associated with it.

Reply:
Moved in: [685dcfc](https://github.com/opensearch-project/security/pull/4992/commits/685dcfc8b7752fc4b118b1b85e3b9e6f827e7c0d)